src/main/java/io/supertokens/config/CoreConfig.java

```diff
@@ -718,16 +767,66 @@ public void createLoggingFile(Main main) throws IOException {
-        // these are all configs that are per core. So we do not allow the developer to set these dynamically.
+        // these are all configs that are per core. So we do not allow the developer to
+        // set these dynamically.
         for (Field field : CoreConfig.class.getDeclaredFields()) {
             if (field.isAnnotationPresent(ConfigYamlOnly.class)) {
                 if (config.has(field.getName())) {
-                    throw new InvalidConfigException(field.getName() + " can only be set via the core's base config setting");
+                    throw new InvalidConfigException(
+                            field.getName() + " can only be set via the core's base config setting");
                 }
             }
         }
     }
 
+    public static JsonArray getConfigFieldsJson() {
+        JsonArray result = new JsonArray();
+
+        for (String fieldId : CoreConfig.getValidFields()) {
+            try {
+                Field field = CoreConfig.class.getDeclaredField(fieldId);
+                // If fieldId exists in PROTECTED_CONFIGS or is not annotated with JsonProperty
+                // or is annotated with ConfigYamlOnly, then skip
+                if (Arrays.asList(PROTECTED_CONFIGS).contains(fieldId) || !field.isAnnotationPresent(JsonProperty.class)
+                        || field.isAnnotationPresent(ConfigYamlOnly.class)) {
+                    continue;
+                }
+                JsonObject fieldJson = new JsonObject();
+                fieldJson.addProperty("name", field.getName());
+                fieldJson.addProperty("description", field.isAnnotationPresent(ConfigDescription.class)
+                        ? field.getAnnotation(ConfigDescription.class).value()
+                        : "");
+                fieldJson.addProperty("isDifferentAcrossTenants",
+                        field.isAnnotationPresent(DifferentAcrossTenants.class));
+
+                String type = "string";
+
+                Class<?> fieldType = field.getType();
+
+                if (fieldType == java.lang.String.class) {
+                    type = "string";
+                } else if (fieldType == boolean.class) {
+                    type = "boolean";
+                } else if (fieldType == int.class || fieldType == long.class || fieldType == double.class) {
+                    type = "number";
+                }
+
+                if (field.isAnnotationPresent(EnumProperty.class)) {
+                    type = "enum";
+                    fieldJson.add("options",
+                            new GsonBuilder().create().toJsonTree(field.getAnnotation(EnumProperty.class).value()));
+                }
+
+                fieldJson.addProperty("type", type);
+                result.add(fieldJson);
+
+            } catch (NoSuchFieldException e) {
+                continue;
+            }
+        }
+        return result;
+    }
+
```

Comment:
in the normalizeAndValidate function in this file, see if you can use the EnumProperty annotation for the validation.

Reply:
Yeah added checks for that in the method

Reply:
I don't see this change